bus/bus.go

```diff
@@ -532,8 +532,21 @@ func (b *Bus) Shutdown(ctx context.Context) error {
 	)
 }
 
-func (b *Bus) addContract(ctx context.Context, rev rhpv2.ContractRevision, contractPrice, totalCost types.Currency, startHeight uint64, state string) (api.ContractMetadata, error) {
-	c, err := b.ms.AddContract(ctx, rev, contractPrice, totalCost, startHeight, state)
+func (b *Bus) addContract(ctx context.Context, rev rhpv2.ContractRevision, contractPrice, initialRenterFunds types.Currency, startHeight uint64, state string) (api.ContractMetadata, error) {
```

Comment:
I considered that but I like that `addContract` will error out with a `DUPLICATE KEY` error if it's being misused. I considered returning a `bool` with `PutContract` and check affected rows but didn't really mind having both `InsertContract` and `PutContract` on the store.

Reply:
I think both solutions would be nicer than duplicate logic.

1. ignoring the issue and use `PutContract` since it's theoretically impossible for a newly formed contract to already exist
2. adding a `force` flag and checking if it was an update or insert

There might even be a variation of 1 where you just try to fetch the contract with `Contract` before calling `PutContract`. As a sanity check and to avoid `force`. Because even though it's not the same db txn, nothing else should be able to accidentally form the same contract in the meantime and insert it.

Reply:
Kept it simple and just removed `InsertContract` and used `PutContract`. It took some strength not to rename it to `UpsertContract` on the `Metadata` store because would be what I would call it personally.